{"cells":[{"cell_type":"markdown","metadata":{"id":"HjdFAOdVxMGd"},"source":["# \"Human upgrade\"\n","\n","Project #5\n","\n","*Lab Journal by Artem Vasilev and Tatiana Lisitsa*\n","\n","---"]},{"cell_type":"markdown","metadata":{"id":"L6zhmfk4OtXM"},"source":["## **Preparing**"]},{"cell_type":"markdown","metadata":{"id":"u7yPElPLPFx4"},"source":["Update packages:\n","\n","`$ sudo apt update && sudo apt upgrade`"]},{"cell_type":"markdown","source":["### **VE setup and downloading tools**"],"metadata":{"id":"eEO0dUvtcfO_"}},{"cell_type":"markdown","source":["Create mamba VE:\n","\n","`$ mamba create -n bi_practice_5 -c conda-forge python=3.12`\n","\n","`$ mamba activate bi_practice_5`"],"metadata":{"id":"pWiujkCfdGIu"}},{"cell_type":"markdown","source":["Install open-cravat:\n","\n","`$ pip install open-cravat`\n","\n","`$ oc config md {path/to/dir/with/annotators}`  # optional\n","\n","`$ oc module install-base`\n","\n","`$ oc module ls -a -t annotator`  # see available annotators\n","\n","`$ oc module install hg19 dbsnp gnomad3 clinvar`"],"metadata":{"id":"Iv22kWka461y"}},{"cell_type":"markdown","source":["Create folder for this project:\n","\n","`$ mkdir Project_5`\n","\n","`$ cd Project_5`"],"metadata":{"id":"vnxtj8lm5Px8"}},{"cell_type":"markdown","source":["Install plink 1.9:\n","\n","`$ mkdir tools && cd tools`\n","\n","`$ wget https://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20231211.zip`\n","\n","`$ unzip plink_linux_x86_64_20231211.zip -d plink`\n","\n","`$ rm plink_linux_x86_64_20231211.zip`\n","\n","`$ cd ..`"],"metadata":{"id":"B7Iap5a6c16u"}},{"cell_type":"markdown","source":["---"],"metadata":{"id":"wY9qBwBY8Q7F"}},{"cell_type":"markdown","source":["## **Data processing**"],"metadata":{"id":"8rCL0wb3etNv"}},{"cell_type":"markdown","source":["### **Downloading**"],"metadata":{"id":"yhF7rak48C_C"}},{"cell_type":"markdown","source":["Download manually raw 23andMe data from [here](https://drive.google.com/file/d/1QJkwJe5Xl_jSVpqdTSNXP7sqlYfI666j/view) and Genotek data from [here](https://drive.google.com/file/d/1piKi0FxVao-IvG81TAXhDKlHXvHE6uTm/view)\n","\n","`$ mkdir data`\n","\n","`$ mv ~/Downloads/genotek.vcf ~/path/to/dir/data/`  # Genotek\n","\n","`$ mv ~/Downloads/SNP_raw_v4_Full_20170514175358.txt.zip ~/path/to/dir/data/`  # 23andME\n","\n","`$ cd ~/path/to/dir/data/`\n","\n","`$ unzip rm SNP_raw_v4_Full_20170514175358.txt.zip`\n","\n","`$ rm SNP_raw_v4_Full_20170514175358.txt.zip`\n","\n","`$ cd ..`"],"metadata":{"id":"ezmmKSBqeyX6"}},{"cell_type":"markdown","source":["### **Preparing for annotation tools**"],"metadata":{"id":"MKCEa4k28Hmv"}},{"cell_type":"markdown","source":["#### **plink**"],"metadata":{"id":"I28GhUgk9T4K"}},{"cell_type":"markdown","source":["Remove all SNPs corresponding to deletions and insertions to make the file compatible with annotation tools\n","\n","`$ mkdir -p results/plink`\n","\n","```bash\n","$ tools/plink/plink \\\n","--23file data/SNP_raw_v4_Full_20170514175358.txt \\\n","--recode vcf \\\n","--out results/plink/snps_clean \\\n","--output-chr MT \\\n","--snps-only just-acgt\n","```\n","\n","The resulting file contains all the analyzed SNPs, and we are interested only in **variable positions**"],"metadata":{"id":"PwJGgB22c2GX"}},{"cell_type":"markdown","source":["#### **open-cravat**"],"metadata":{"id":"jJMISprS9Wkz"}},{"cell_type":"markdown","source":["\n","`$ oc gui `\n","\n","With gui at the store we can install module to use 23andMe .txt file as input without any conversion\n","\n","\n","`$ oc run SNP_raw_v4_Full_20170514175358.txt -l hg19 -t excel`"],"metadata":{"id":"8VQWD9FM9XK9"}},{"cell_type":"markdown","source":["---"],"metadata":{"id":"Ibqize4oEFR2"}},{"cell_type":"markdown","source":["## **Data analyzing**"],"metadata":{"id":"4E2GbEjX7EaF"}},{"cell_type":"markdown","source":["### **Origin and haplogroups**"],"metadata":{"id":"oaBRNRXXc2MI"}},{"cell_type":"markdown","source":["Web-tool [mthap](https://dna.jameslick.com/mthap/)\n","\n","Choose raw 23andMe file: `SNP_raw_v4_Full_20170514175358.txt`\n","\n","Interpret the results ([FAQ](https://dna.jameslick.com/mthap/FAQ.html)):\n","\n","- \"Match\" is a marker for which you match what is required for this haplogroup.\n","- \"Mismatch\" is a marker required for the haplogroup for which you have a different test result. This can indicate either an inaccurate test result, a reversion for the marker or that you are a poor match for the haplogroup. It may also indicate that you are \"in between\" two different haplogroups.\n","- An \"Extra\" is a marker you have that is not part of the defining markers list for the haplogroup. These could be private mutations, an indication of a poor match, or an inaccurate test result.\n","- \"No-Call\" means that the marker was tested but, for whatever reason, it was not possible to determine the result.\n","- \"Untested\" means that your testing service did not test this position. Usually there are enough other markers to determine a match, but a long list of Untested markers reduces the accuracy of the match."],"metadata":{"id":"8PrwWDtqc2Oe"}},{"cell_type":"markdown","source":["### **Eye color**"],"metadata":{"id":"pFlGn4aPc2Q0"}},{"cell_type":"markdown","source":["For eye colour identification we used this [article](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3694299/):"],"metadata":{"id":"M-NCbxmy-UQy"}},{"cell_type":"markdown","source":["![](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3694299/bin/CroatMedJ_54_0248-F2.jpg)"],"metadata":{"id":"8H-Zyl0o7NPH"}},{"cell_type":"markdown","source":["`$ cat results/plink/snps_clean.vcf | grep rs12913832` genotype A/G: → Not Blue\n","\n","---\n","\n","`$ cat results/plink/snps_clean.vcf | grep rs16891982` genotype C/G: → Not Brown\n","\n","`$ cat results/plink/snps_clean.vcf | grep rs6119471` NA\n","\n","---\n","\n","`$ cat results/plink/snps_clean.vcf | grep rs12203592` genotype C/T: → Not Green\n","\n","`$ cat results/plink/snps_clean.vcf | grep rs16891982` genotype C/G: → Not Green"],"metadata":{"id":"wK1F80UqCS1A"}},{"cell_type":"markdown","source":["### **Annotation of all SNPs, selection of clinically relevant ones**"],"metadata":{"id":"43HtqxBpC9D2"}},{"cell_type":"markdown","source":["#### **[VEP](https://grch37.ensembl.org/Homo_sapiens/Tools/VEP) (Variant Effect Predictor)**"],"metadata":{"id":"jbcn-74GDAG9"}},{"cell_type":"markdown","source":["Assembly: <u>GRCh37.p13</u>\n","\n","Choose file: `snps_clean`\n","\n","---\n","\n","Command line equivalent of job:\n","\n","`$ ./vep --af --af_gnomade --af_gnomadg --appris --biotype --buffer_size 500 --canonical --check_existing --distance 5000 --hgvs --mane --merged --numbers --plugin MPC,[path_to]/fordist_constraint_official_mpc_values.txt.gz --plugin LOEUF,file=[path_to]/loeuf_dataset_grch37.tsv.gz,match_by=gene --plugin CADD,[path_to]/CADD_GRCh37_1.6_whole_genome_SNVs.tsv.gz,[path_to]/CADD_GRCh37_1.6_InDels.tsv.gz --plugin SpliceAI,snv=[path_to]/spliceai_scores.masked.snv.hg19.vcf.gz,indel=[path_to]/spliceai_scores.masked.indel.hg19.vcf.gz --plugin Phenotypes,dir=[path_to]/,phenotype_feature=1,exclude_sources=COSMIC&HGMD-PUBLIC&Cancer_Gene_Census --plugin AncestralAllele,[path_to]/homo_sapiens_ancestor_GRCh37_e75.fa.gz --plugin IntAct,mapping_file=[path_to]/mutation_gc_map.txt.gz,mutation_file=[path_to]/mutations.tsv,pmid=1 --polyphen b --protein --pubmed --regulatory --show_ref_allele --sift b --species homo_sapiens --symbol --transcript_version --tsl --uploaded_allele --cache --input_file [input_data] --output_file [output_file] --port 3337`"],"metadata":{"id":"4Tplu8iEDRx5"}},{"cell_type":"markdown","source":["##### **Download and move result files to working directory:**"],"metadata":{"id":"O8B--sr9q4Bq"}},{"cell_type":"markdown","source":["`$ mkdir -p results/vec`\n","\n","`$ mv ~/Downloads/<vec_result_file.vcf> ~/path/to/dir/results/vec`\n","\n","`$ mv ~/Downloads/<vec_result_file.txt> ~/path/to/dir/results/vec`"],"metadata":{"id":"UuF7Qsj3ET6y"}},{"cell_type":"markdown","source":["##### **Filter pathogenic variants:**"],"metadata":{"id":"7Zszl9fDq6sy"}},{"cell_type":"markdown","source":["`$ cat ~/path/to/dir/results/vec/<vec_result_file.vcf> | grep pathogenic | grep -wv benign > ~/path/to/dir/results/vec/pathogenic.vcf`\n","\n","Example: `$ cat results/vec/j9zxJlHDXeHehG3F.vcf | grep pathogenic | grep -wv benign > results/vec/pathogenic.vcf`\n","\n","**i5005436**\n","\n","https://franklin.genoox.com/clinical-db/variant/snp/chr6-32008312-C-T?app=assessment-tools\n","\n","---\n","\n","**rs2004640**\n","\n","https://franklin.genoox.com/clinical-db/variant/snp/chr7-128578301-G-T?app=assessment-tools"],"metadata":{"id":"SQgrFd4zvetg"}},{"cell_type":"markdown","source":["\n","##### **Filter pathogenic variants w/o introns:**"],"metadata":{"id":"1XfweeCcrD9t"}},{"cell_type":"markdown","source":["`$ cat ~/path/to/dir/results/vec/<vec_result_file.vcf> | grep pathogenic | grep -wv benign | grep -wv intron_variant > ~/path/to/dir/results/vec/pathogenic_wo_intron.vcf`\n","\n","Example:\n","\n","**rs1024611**\n","\n","https://franklin.genoox.com/clinical-db/variant/snp/chr17-32579788-A-G?app=assessment-tools\n","\n","---\n","\n","**i5005436** - already found\n","\n","https://franklin.genoox.com/clinical-db/variant/snp/chr6-32008312-C-T?app=assessment-tools"],"metadata":{"id":"QHkAah06vfjw"}},{"cell_type":"markdown","source":["#### **Processing Open-cravat output**"],"metadata":{"id":"glWWIwk6rHL5"}},{"cell_type":"markdown","source":["```{python}\n","import pandas as pd\n","input_file = '~/path/to/dir/results//SNP_raw_v4_Full_20170514175358.txt.xlsx'\n","sheet_name = 'Variant'\n","header_row = 1\n","df = pd.read_excel(input_file, sheet_name=sheet_name, header=header_row)\n","csv_file_path = '23andMe.csv'\n","df.to_csv(csv_file_path, index=False)\n","\n","df_23andMe = pd.read_csv(r'23andMe.csv')\n","df_23andMe.shape()\n","\n","# Filtered Global AF, Clinical Significance\n","df_23andMe = df_23andMe[((df_23andMe['Global AF'].isna()) | (df_23andMe['Global AF'] < 0.05)) & ((df_23andMe['Clinical Significance'].notnull()) & (df_23andMe['Clinical Significance'].apply(lambda val: all(val != s for s in ['Benign', 'Benign/Likely benign', 'Likely benign', 'Uncertain significance', 'drug response', 'not provided', 'Conflicting interpretations of pathogenicity', 'other']))))]\n","```\n","\n","Then we can directly read about this varian on Franklin"],"metadata":{"id":"B58-MbpfTYch"}}],"metadata":{"colab":{"provenance":[],"toc_visible":true},"kernelspec":{"display_name":"Python 3 (ipykernel)","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.12"}},"nbformat":4,"nbformat_minor":0}